### PR TITLE
Improving complexity calculation

### PR DESCRIPTION
**Pull request by pete-proton**
no need for list of integers for storing complexity - integer as a sum is good enough

### CODE REVIEW

File: src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java [hunk 1]
```diff
@@ -54,20 +54,21 @@ public MaxQueryComplexityInstrumentation(int maxComplexity, FieldComplexityCalcu
...
-                    if (valuesByParent.containsKey(env)) {
-                        childsComplexity = valuesByParent.get(env).stream().mapToInt(Integer::intValue).sum();
-                    }
+                    int childsComplexity = valuesByParent.getOrDefault(env, 0);
                     int value = calculateComplexity(env, childsComplexity);
-                    valuesByParent.putIfAbsent(env.getParentEnvironment(), new ArrayList<>());
-                    valuesByParent.get(env.getParentEnvironment()).add(value);
+
+                    valuesByParent.compute(env.getParentEnvironment(), (key, oldValue) ->
+                        ofNullable(oldValue).orElse(0) + value
```
Comment (bbakerman) [inline src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java]: This sets it to `oldValue + value` and not just `value` like it did before

**Comment by pete-proton** [comment]
raised this small one 2 weeks ago - just one reply, which seemed to me like the PR was not looked at thoroughly. could you tell me if there is anything needs to be done from you PoV? 

TBH feel a little discouraged raising further PR's

**Comment by andimarek** [comment]
@pete-proton I am sorry and I can understand your reaction, but this a private run project with limited time we can spend on it. This means unfortunately it can take quite some time until we get back to PRs, especially not urgent PRs.

We really appreciate your contribution and I hope we see more in the future.

Thanks,
Andi

File: src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java [hunk 2]
```diff
@@ -88,11 +89,11 @@ protected AbortExecutionException mkAbortException(int totalComplexity, int maxC
 
     QueryTraversal newQueryTraversal(InstrumentationValidationParameters parameters) {
         return QueryTraversal.newQueryTraversal()
-                .schema(parameters.getSchema())
-                .document(parameters.getDocument())
-                .operationName(parameters.getOperation())
-                .variables(parameters.getVariables())
-                .build();
+                             .schema(parameters.getSchema())
```
Comment (andimarek) [inline src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java]: can you make sure you are using the https://github.com/graphql-java/graphql-java/blob/master/graphql-java-code-style.xml codestyle? thanks

**Comment by pete-proton** [comment]
thank you! do I need to reformat still? sorry for the late reply

**Comment by bbakerman** [comment]
If you could then that would be great

**Comment by pete-proton** [comment]
sure, sorry took a while, but finally created https://github.com/graphql-java/graphql-java/pull/1323

**Pull request merged**